atomic_reactor/plugins/post_pulp_sync.py

```diff
@@ -170,22 +167,28 @@ def run(self):
         self.log.info("syncing from docker V2 registry %s",
                       self.docker_registry)
 
+        docker_registry = hostname_and_port.sub(lambda m: m.groups()[0],
+                                                self.docker_registry)
+
+        kwargs = self.get_dockercfg_credentials(docker_registry)
+        if self.insecure_registry is not None:
+            kwargs['ssl_validation'] = not self.insecure_registry
+
         images = []
         repos = {}  # pulp repo -> repo id
-        with dockpulp_config(docker_registry=self.docker_registry) as config:
-            for image in self.workflow.tag_conf.primary_images:
-                if image.pulp_repo not in repos:
-                    self.log.info("syncing %s", image.pulp_repo)
-                    repoinfo = pulp.syncRepo(config.env, image.pulp_repo,
-                                             config_file=config.name)
-                    repos[image.pulp_repo] = repoinfo[0]['id']
-
+        for image in self.workflow.tag_conf.primary_images:
+            if image.pulp_repo not in repos:
+                self.log.info("syncing %s", image.pulp_repo)
+                repoinfo = pulp.syncRepo(feed=self.docker_registry,
```

Comment:
I added a commit for this. I left the V1 Pulp code alone  in order to avoid destabilising anything there.